File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JTextFieldWrapper.java
package org.codinjutsu.tools.nosql.commons.view.add;

import javax.swing.JTextField;

import static org.apache.commons.lang.StringUtils.isNotBlank;

public abstract class JTextFieldWrapper extends TextFieldWrapper<JTextField> {

    JTextFieldWrapper() {
        super(new JTextField());
    }

    @Override
    public boolean isValueSet() {
        return isNotBlank(component.getText());
    }

    @Override
    public void reset() {
        component.setText("");
    }
}

File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JTextFieldWrapper.kt
package org.codinjutsu.tools.nosql.commons.view.add

import javax.swing.JTextField

internal abstract class JTextFieldWrapper internal constructor() : AbstractTextFieldWrapper<JTextField>(JTextField()) {

    override val isValueSet: Boolean
        get() = component.text.isNotBlank()

    override fun reset() {
        component.text = ""
    }
}
